admob/src/android/banner_view_internal_android.cc

```diff
@@ -192,96 +232,173 @@ void InitializeBannerViewOnMainThread(void* data) {
                       ad_listener);
   jni_exception = util::CheckAndClearJniExceptions(env);
   FIREBASE_ASSERT(!jni_exception);
+
+  env->CallVoidMethod(call_data->ad_view,
+                      ad_view::GetMethodId(ad_view::kSetOnPaidEventListener),
+                      ad_listener);
+  jni_exception = util::CheckAndClearJniExceptions(env);
+  FIREBASE_ASSERT(!jni_exception);
+
   env->DeleteLocalRef(ad_listener);
 
   call_data->callback_data->future_data->future_impl.Complete(
       call_data->callback_data->future_handle, kAdMobErrorNone, "");
 
+  delete call_data->callback_data;
+  call_data->callback_data = nullptr;
   delete call_data;
 }
 
 Future<void> BannerViewInternalAndroid::Initialize(AdParent parent,
                                                    const char* ad_unit_id,
                                                    const AdSize& size) {
-  FutureCallbackData<void>* callback_data =
-      CreateVoidFutureCallbackData(kBannerViewFnInitialize, &future_data_);
-  SafeFutureHandle<void> future_handle = callback_data->future_handle;
+  firebase::MutexLock lock(mutex_);
 
   if (initialized_) {
+    const SafeFutureHandle<void> future_handle =
+        future_data_.future_impl.SafeAlloc<void>(kBannerViewFnInitialize);
     CompleteFuture(kAdMobErrorAlreadyInitialized,
                    kAdAlreadyInitializedErrorMessage, future_handle,
                    &future_data_);
-  } else {
```

Comment:
remove the else clause so this looks like a major deletion. It's not, it appears again below.